src/azul/portal_service.py

```diff
@@ -172,12 +173,17 @@ def _crud(self, operation: Callable[[JSONs], Optional[JSONs]]) -> Optional[JSONs
                     except NoSuchObjectVersion:
                         # Wait for latest version to appear in S3.
                         continue
-                db = operation(db)
-                if db is None:
-                    # Operation is read-only; we're done.
+                new_db = operation(db)
+                if db == new_db:
+                    # Operation didn't modify DB; we're done.
                     break
+                elif new_db is None:
+                    # FIXME: Add delete logic for crud
+                    #        https://github.com/DataBiosphere/azul/issues/3484
+                    pass
```

Comment:
Consider changing this to 
```suggestion
                    assert False
```
since passing a callback that returns `None` currently is not supported and contradicts the type annotation.